python/ray/utils.py

```diff
@@ -276,23 +276,13 @@ def setup_logger(logging_level, logging_format):
         logging_level = logging.getLevelName(logging_level.upper())
     logger.setLevel(logging_level)
     global _default_handler
-    _default_handler = logging.StreamHandler()
+    if _default_handler is None:
```

Comment:
@richardliaw please take a look. This is a bug fix. Before we were adding a new handler every time `setup_logger` was called, so if you did

```python
while True:
    ray.init()
    ray.shutdown()
```

the number of handlers would grow unbounded and logging statements would be replicated many times.

Reply:
lgtm